generator/road_access_generator.cpp

```diff
@@ -0,0 +1,162 @@
+#include "generator/road_access_generator.hpp"
+
+#include "generator/osm_element.hpp"
+#include "generator/routing_helpers.hpp"
+
+#include "routing/road_access.hpp"
+#include "routing/road_access_serialization.hpp"
+
+#include "indexer/classificator.hpp"
+#include "indexer/feature_data.hpp"
+
+#include "coding/file_container.hpp"
+#include "coding/file_writer.hpp"
+
+#include "base/logging.hpp"
+#include "base/scope_guard.hpp"
+#include "base/string_utils.hpp"
+
+#include "std/initializer_list.hpp"
+
+#include "defines.hpp"
+
+#include <algorithm>
+#include <fstream>
+#include <map>
+#include <string>
+
+using namespace std;
+
+namespace
+{
+char const kAccessPrivate[] = "access=private";
+char const kBarrierGate[] = "barrier=gate";
+char const kDelim[] = " \t\r\n";
+}  // namespace
+
+namespace routing
+{
+// RoadAccessWriter ------------------------------------------------------------
+
+void RoadAccessWriter::Open(string const & filePath)
+{
+  LOG(LINFO,
+      ("Saving information about barriers and road access classes in osm id terms to", filePath));
+  m_stream.open(filePath, ofstream::out);
+
+  if (!IsOpened())
+    LOG(LINFO, ("Cannot open file", filePath));
+}
+
+void RoadAccessWriter::Process(OsmElement * p, FeatureParams & params)
+{
+  if (!IsOpened())
+  {
+    LOG(LWARNING, ("Tried to write to a closed barriers writer"));
+    return;
+  }
+
+  auto const & c = classif();
+
+  StringIL const forbiddenRoadTypes[] = {
+    {"hwtag", "private"}
+  };
+
+  for (auto const & f : forbiddenRoadTypes)
+  {
+    auto const t = c.GetTypeByPath(f);
+    if (params.IsTypeExist(t) && p->type == OsmElement::EntityType::Way)
+      m_stream << kAccessPrivate << " " << p->id << "\n";
```

Comment:
But you violate system-default newline symbol. On some systems it's "\r\n", not just "\n".